File: c++/0070.cpp
// Link: https://leetcode.com/problems/climbing-stairs/description/
// Runtime: 0 ms
// Memory: 6.47 MB

class Solution {
public:
  int climbStairs(int n) {
    long long oneBefore = 1, twoBefore = 1;
    for (int i = 0; i < n; i++) {
      long long temp = oneBefore;
      oneBefore += twoBefore;
      twoBefore = temp;
    }

    return twoBefore;
  }
};

File: python/0070.py
# Link: https://leetcode.com/problems/climbing-stairs/description/
# Runtime: 34 ms
# Memory: 17.35 MB


class Solution:
    @cache
    def climbStairs(self, n: int) -> int:
        if n < 3:
            return n

        return self.climbStairs(n - 1) + self.climbStairs(n - 2)
